src/main/java/org/broadinstitute/hellbender/utils/read/ReadCoordinateComparator.java

```diff
@@ -54,10 +60,9 @@ private int compareCoordinates( final GATKRead first, final GATKRead second ) {
         final int firstRefIndex = ReadUtils.getReferenceIndex(first, header);
         final int secondRefIndex = ReadUtils.getReferenceIndex(second, header);
 
-        if ( first.isUnmapped() ) {
-            return second.isUnmapped() ? 0 : 1;
-        }
-        else if ( second.isUnmapped() ) {
+        if ( firstRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ) {
+            return secondRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ? 0 : 1;
+        } else if ( secondRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ) {
```

Comment:
We do not ever consider reads with reference index -1 as "mapped" in hellbender -- this violates the contract of the `GATKRead` interface, which considers all such reads as unmapped. Since this is a comparator for GATKReads, not SAMRecords, we should be using GATKRead.isUnmapped() instead of looking at the raw reference indices. Also note that with headerless reads in Spark, we potentially have only a reference name, and no reference index, and this change will cause incorrect results in that case.